webv2/profile/profile.go

```diff
@@ -9,19 +9,32 @@ import (
 	"net/http"
 	"strconv"
 	"strings"
+	"sync"
 
 	datastream "cloud.google.com/go/datastream/apiv1"
+	ds "github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/clients/datastream"
 	storageclient "github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/clients/storage"
+	datastream_accessor "github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/datastream"
 	storageaccessor "github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/storage"
 	"github.com/GoogleCloudPlatform/spanner-migration-tool/common/constants"
 	"github.com/GoogleCloudPlatform/spanner-migration-tool/common/utils"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/conversion"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/profiles"
 	"github.com/GoogleCloudPlatform/spanner-migration-tool/streaming"
 	"github.com/GoogleCloudPlatform/spanner-migration-tool/webv2/helpers"
 	"github.com/GoogleCloudPlatform/spanner-migration-tool/webv2/session"
 	"google.golang.org/api/iterator"
 	datastreampb "google.golang.org/genproto/googleapis/cloud/datastream/v1"
 )
 
+type shardedDataflowConfig struct {
+	MigrationProfile profiles.SourceProfileConfig
+}
```

Comment:
Why is this defined again? We should have something similar already?

Reply:
It reuses profiles.SourceProfileConfig. The other place where profiles.SourceProfileConfig is used is in SourceProfile:
```
type SourceProfile struct {
	Driver string
	Ty     SourceProfileType
	File   SourceProfileFile
	Conn   SourceProfileConnection
	ConnCloudSQL SourceProfileConnectionCloudSQL
	Config SourceProfileConfig
	Csv    SourceProfileCsv
}
```

I felt reusing that doesn't fit right